core/server/worker/src/test/java/alluxio/worker/page/PagedBlockStoreCommitBlockTest.java

```diff
@@ -0,0 +1,278 @@
+    } catch (Exception e) {
+      throw e;
+    }
+
+    mListener = mock(BlockStoreEventListener.class);
+    doAnswer((i) -> {
+      assertEquals(BLOCK_ID, i.getArguments()[0]);
+      return 0;
+    }).when(mListener).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+    doAnswer((i) -> {
+      assertEquals(BLOCK_ID, i.getArguments()[0]);
+      return 0;
+    }).when(mListener).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+  }
+
+  // This Test case success both to commit, no Exception should be thrown,
+  // and both onCommit method should be called
+  @Test
+  public void localCommitAndMasterCommitBothSuccess()
+         throws IOException, InterruptedException, TimeoutException {
+    try {
+      mPageMetaStore = new PagedBlockMetaStore(mDirs) {
+        // here commit always success
+        @Override
+        public PagedBlockMeta commit(long BLOCK_ID) {
+            return super.commit(BLOCK_ID);
+        }
+      };
+      mCacheManager = CacheManager.Factory.create(mConf, mCacheManagerOptions, mPageMetaStore);
+
+      mPagedBlockStore = new PagedBlockStore(mCacheManager, mUfs, mBlockMasterClientPool, mWorkerId,
+              mPageMetaStore, mCacheManagerOptions.getPageSize());
+    } catch (IOException e) {
+      throw new RuntimeException();
+    }
+
+    try {
+      prepareBlockStore();
+    } catch (Exception e) {
+      throw e;
+    }
+
+    mPagedBlockStore.commitBlock(SESSION_ID, BLOCK_ID, false);
+    verify(mListener).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+    verify(mListener).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+  }
+
+  // This Test case success commitToMaster, expecting one exception,
+
+  @Test
+  public void localCommitFailAndMasterCommitSuccess()
+          throws IOException, InterruptedException, TimeoutException {
+    try {
+      mPageMetaStore = new PagedBlockMetaStore(mDirs) {
+          // here commit always throw Exception
+          @Override
+          public PagedBlockMeta commit(long BLOCK_ID) {
+              throw new RuntimeException();
+          }
+      };
+      mCacheManager = CacheManager.Factory.create(mConf, mCacheManagerOptions, mPageMetaStore);
+
+      mPagedBlockStore = new PagedBlockStore(mCacheManager, mUfs, mBlockMasterClientPool,
+              mWorkerId, mPageMetaStore, mCacheManagerOptions.getPageSize());
+    } catch (IOException e) {
+      throw new RuntimeException();
+    }
+
+    try {
+      prepareBlockStore();
+    } catch (Exception e) {
+      throw e;
+    }
+
+    assertThrows(RuntimeException.class, () -> {
+      mPagedBlockStore.commitBlock(SESSION_ID, BLOCK_ID, false);
+    });
+
+    verify(mListener, never()).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+    verify(mListener, never()).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+
+    try {
+      mPagedBlockStore.close();
```

Comment:
actually it's OK to not close the store in a unit test. If you need to, you can add a `@After` method and put all closers in that method.